app/services/document_converter_service.rb

```diff
@@ -0,0 +1,43 @@
+class DocumentConverterService
+  def initialize(original, converted)
+    @original = original
+    @converted = converted
+  end
+
+  def call
+    return if @converted.attached?
+
+    if @original.content_type == 'application/pdf'
+      @converted.attach(@original.blob)
+    else
+      convert_to_pdf from: @original, to: @converted
+    end
+  end
+
+  private
+
+  def convert_to_pdf(from:, to:)
+    with_attached_file(from) do |original|
+      pdf_tmpfile = Tempfile.new
+      Libreconv.convert(original, pdf_tmpfile)
+      to.attach(io: pdf_tmpfile, filename: "#{from.filename}.pdf", content_type: 'application/pdf')
+    end
+  rescue IOError
+    nil # raised if Libreoffice exe is not in PATH
```

Comment:
rather see a log error than a comment, if necessary

Reply:
Thanks. This was copied verbatim from the existing code but the rescue can actually be removed as we want it to raise the exception so that the job will be retried.